src/Formatter/ProtobufFormatter.php

```diff
@@ -0,0 +1,352 @@
+<?php
+
+namespace Crell\Serde\Formatter;
+
+use Crell\Serde\Attributes\Binary;
+use Crell\Serde\Attributes\ClassSettings;
+use Crell\Serde\Attributes\Field;
+use Crell\Serde\Attributes\Fixed32;
+use Crell\Serde\Attributes\Fixed64;
+use Crell\Serde\Attributes\Float32;
+use Crell\Serde\Attributes\Float64;
+use Crell\Serde\Attributes\Int32;
+use Crell\Serde\Attributes\Int64;
+use Crell\Serde\Attributes\SFixed32;
+use Crell\Serde\Attributes\SFixed64;
+use Crell\Serde\Attributes\SInt32;
+use Crell\Serde\Attributes\SInt64;
+use Crell\Serde\Attributes\UInt32;
+use Crell\Serde\Attributes\UInt64;
+use Crell\Serde\Dict;
+use Crell\Serde\Formatter\Protobuf\ProtoWireType;
+use Crell\Serde\Formatter\Protobuf\RunningValue;
+use Crell\Serde\Sequence;
+use Crell\Serde\Serializer;
+
+class ProtobufFormatter implements Formatter
+{
+    private const FULL32 = 0xffffffff;
+    private const INT32_MAX = '4294967296';
```

Comment:
```suggestion
    private const Full32 = 0xffffffff;
    private const Int32Max = '4294967296';
```

(And wherever those are used.)

Yes that's not standard, but I find PascalCase constants far easier to work with.